File: Python/844_BackspaceStringCompare.py
class Solution:
    def backspaceCompare(self, S: str, T: str) -> bool:
        sS = []
        sT = []
        for i in range(0, len(S)):
            if S[i] != '#':
                sS.append(S[i])
            elif len(sS) > 0:
                sS.pop()
        for i in range(0, len(T)):
            if T[i] != '#':
                sT.append(T[i])
            elif len(sT) > 0:
                sT.pop()
            
        if sS == sT:
            return True
        else:
            return False
        

File: C#/844_BackspaceStringCompare.cs
public class Solution {
    public bool BackspaceCompare(string S, string T) {
        Stack<char> nS = new Stack<char>();     
        Stack<char> nT = new Stack<char>();    
        foreach(char c in S){
            if (c != '#')
                nS.Push(c);
            else if (nS.Count > 0)
                nS.Pop();
        }
        
        foreach(char c in T){
            if (c != '#')
                nT.Push(c);
            else if (nT.Count > 0)
                nT.Pop();
        }
        
        if (nS.Count != nT.Count) return false;
        
        while(nS.Count > 0) {
            if (nS.Pop() != nT.Pop())
                return false;
        }
        
        return true;
    }
}
